### PR TITLE
Add artifacts for a secondary (user) node pool

### PR DESCRIPTION
**Pull request by heoelri**
This PR adds a commented out secondary node pool to keep the main / default node pool for "system" only and add a dedicated "user" node pool for all workload components. The secondary node pool will be commented out and kept in code only as an example.

image

https://github.com/Azure/Mission-Critical-Online/blob/40ff0031ce59f19980cb30fe78ef284c0c69879d/src/infra/workload/releaseunit/modules/stamp/kubernetes.tf#L75-L96

image

https://github.com/Azure/Mission-Critical-Online/blob/40ff0031ce59f19980cb30fe78ef284c0c69879d/src/app/charts/catalogservice/templates/deployment.yaml#L66-L73

### CODE REVIEW

File: src/infra/README.md [hunk 1]
```diff
@@ -179,21 +179,22 @@ The current networking setup consists of a single Azure Virtual Network per _sta
 
 Azure Kubernetes Service (AKS) is used as the compute platform as it is most versatile and as Kubernetes is the de-facto compute platform standard for modern applications, both inside and outside of Azure.
 
-Azure Mission-Critical uses Linux-only clusters as there is no requirement for any Windows-based containers and Linux is the more mature platform in terms of Kubernetes.
+This Azure Mission-Critical reference implementation uses Linux-only clusters only as its sample workload is written in .NET core and there is no requirement for any Windows-based containers.
```
Comment (msimecek) [inline src/infra/README.md:182]:
```suggestion
This Azure Mission-Critical reference implementation uses Linux-only clusters as its sample workload is written in .NET Core and there is no requirement for any Windows-based containers.
```

File: src/infra/README.md [hunk 2]
```diff
@@ -179,21 +179,22 @@ The current networking setup consists of a single Azure Virtual Network per _sta
...
   - `os_disk_type` is set to `Ephemeral` to leverage [Ephemeral OS disks](https://docs.microsoft.com/azure/aks/cluster-configuration#ephemeral-os) for performance reasons.
   - `upgrade_settings` `max_surge` is set to `33%` which is the [recommended value for production workloads](https://docs.microsoft.com/azure/aks/upgrade-cluster#customize-node-surge-upgrade).
+  > **Important!** In production environments its recommended to separate system and user node pools (see [Manage system node pools in Azure Kubernetes Service](https://docs.microsoft.com/azure/aks/use-system-pools)). This Azure Mission-Critical reference implementation is using a single (system) node pool for cost-saving purposes and to reduce overhead.
+  > The `kubernetes.tf` file contains a commented-out example for an additional user node pool with a taint `workload=true:NoSchedule` set to prevent non-workload pods from being scheduled. The `node_label` set to `role=workload` can be used to target this node pool when deploying a workload (see [charts/catalogservice](/src/app/charts/catalogservice/) for an example).
```
Comment (msimecek) [inline src/infra/README.md:197]: Pardon my ignorance - `NoSchedule` taint means that the workload pod **will** be scheduled?

**Review by msimecek — CHANGES_REQUESTED** [review]

File: .ado/pipelines/templates/stages-full-release.yaml [hunk 3]
```diff
@@ -168,7 +168,7 @@ stages:
                                     -var=queued_by="$(Build.QueuedBy)"
                                     -var=contact_email="$(contactEmail)"
                                     -var=''stamps=$(stampLocations)''
-                                    -var=kubernetes_version="$(kubernetesVersion)"
+                                    -var=aks_kubernetes_version="$(kubernetesVersion)"
```
Comment (sebader) [inline .ado/pipelines/templates/stages-full-release.yaml:171]: aks_kubernetes is somewhat redundant, dont you think? how about aks_version?

File: src/app/AlwaysOn.UI/package.json [hunk 4]
```diff
@@ -19,7 +19,8 @@
     "@vue/cli-service": "^5.0.4",
     "is-primitive": "^3.0.1",
     "nprogress": "^0.2.0",
-    "vue-template-compiler": "^2.6.14"
+    "vue-template-compiler": "^2.6.14",
```
Comment (sebader) [inline src/app/AlwaysOn.UI/package.json:22]: why is this in this PR?

File: src/infra/README.md (hunk 1 above)
Comment (sebader) [inline src/infra/README.md]:
```suggestion
This Azure Mission-Critical reference implementation solely uses Linux-only clusters as its sample workload is written in .NET Core and there is no requirement for any Windows-based containers.
```

File: src/infra/README.md (hunk 2 above)
Comment (heoelri) [reply src/infra/README.md:197]: Yes - the node has a "taint" and the pod/or deployment needs a "toleration". Means a deployment with this toleration will schedule pods on this node(s).

**Review by sebader — CHANGES_REQUESTED** [review]

File: src/app/AlwaysOn.UI/package.json (hunk 4 above)
Comment (msimecek) [reply src/app/AlwaysOn.UI/package.json:22]: I presume adopted from #335 to make it work for Heyko.

File: .ado/pipelines/templates/stages-full-release.yaml (hunk 3 above)
Comment (heoelri) [reply .ado/pipelines/templates/stages-full-release.yaml:171]: I did change this for consistency. All the other "aks" variables had "aks_" as a prefix. And - it's not the aks_version. It's the kubernetes_version used for this specific AKS (node pool - though we're not using different vers for cluster and node pools).

File: src/app/AlwaysOn.UI/package.json (hunk 4 above)
Comment (heoelri) [reply src/app/AlwaysOn.UI/package.json:22]: This comes from PR #335 to make the PR run again. Merged PR #335 into this PR.

File: .ado/pipelines/templates/stages-full-release.yaml (hunk 3 above)
Comment (heoelri) [reply .ado/pipelines/templates/stages-full-release.yaml:171]: image

File: src/app/AlwaysOn.UI/package.json (hunk 4 above)
Comment (sebader) [reply src/app/AlwaysOn.UI/package.json:22]: ok then lets get #335 merged into main and rebase this one :)

File: .ado/pipelines/templates/stages-full-release.yaml (hunk 3 above)
Comment (sebader) [reply .ado/pipelines/templates/stages-full-release.yaml:171]: what other aks version can there be, then? If you dont want to call it aks_version?

Comment (heoelri) [reply .ado/pipelines/templates/stages-full-release.yaml:171]: At some point maybe the AKS Release version:

image

Comment (heoelri) [reply .ado/pipelines/templates/stages-full-release.yaml:171]: Why should this parameter that configures an AKS setting not start with the AKS prefix like the others?

Comment (sebader) [reply .ado/pipelines/templates/stages-full-release.yaml:171]: I'm ok to keep it like this if you think there is potential future use. its just that Azure Kubernetes Service Kubernetes Version felt a bit redundant to me

Comment (heoelri) [reply .ado/pipelines/templates/stages-full-release.yaml:171]: I see your point, but it's not the "aks version" but still a parameter for "aks" so I'd like to have it consistent using a common prefix "aks_" for all aks variables.

**Pull request merged**